cmd/backfill/main.go

```diff
@@ -0,0 +1,330 @@
+}
+
+func main() {
+	cfg := &config{
+		file: os.Getenv(envFile),
+		host: os.Getenv(envHost),
+		auth: os.Getenv(envAuth),
+	}
+	if err := execute(cfg); err != nil {
+		fmt.Fprintf(os.Stdout, "backfill failure: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func execute(cfg *config) error {
+	url := fmt.Sprintf(healthcheck, cfg.host)
+
+	/* #nosec G107 */
+	if _, err := http.Get(url); err != nil {
+		return fmt.Errorf("failed healthcheck for [%s]: %w", url, err)
+	}
+
+	f, err := os.Open(cfg.file)
+	if err != nil {
+		return err
+	}
+	src := csv.NewReader(f)
+	row, err := src.Read()
+	if err != nil {
+		return fmt.Errorf("failed to read column headers for [%s]: %w", cfg.file, err)
+	}
+	// fmt.Fprintf(os.Stdout, "headers: %v\n", row)
+	_ = row
+	rowCount := 0
+	createCount := 0
+	updateCount := 0
+
+	errs := []error{}
+	for err == nil {
+		row, err = src.Read()
+		if err != nil {
+			if err != io.EOF {
+				errs = append(errs, err)
+			}
+			continue
+		}
+		rowCount++
+		item, rErr := convert(row)
+		if rErr != nil {
+			errs = append(errs, rErr)
+			continue
+		}
+		if item == nil {
+			continue
+		}
+		create, rErr := upload(cfg.host, cfg.auth, item)
+		if rErr != nil {
+			errs = append(errs, rErr)
+		}
+		if create {
+			createCount++
+		} else {
+			updateCount++
+		}
+	}
+
+	fmt.Printf("%d rows processed with %d errors.\n%d creates and %d updates.\n\n", rowCount, len(errs), createCount, updateCount)
+
+	if len(errs) != 0 {
+		fmt.Fprintf(os.Stdout, "problems processing file: %v", errs)
+		return fmt.Errorf("problems processing file: %v", errs)
+	}
+	return nil
+}
+
+func upload(host string, auth string, item *entry) (didCreate bool, err error) {
+	body, err := json.MarshalIndent(item, "", "\t")
+	if err != nil {
+		return false, err
+	}
+
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/api/v1/backfill", host), bytes.NewReader(body))
+	if err != nil {
+		return false, err
+	}
+
+	if ok, perr := strconv.ParseBool(os.Getenv(envDrop)); ok && perr == nil {
+		req, err = http.NewRequest("DELETE", fmt.Sprintf("%s/api/v1/system_intake/%s?remove=true", host, item.Intake.ID.String()), nil)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	req.Header.Add("Authorization", auth)
+	// req.Write(os.Stdout)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false, fmt.Errorf("request failed: %v", err)
+	}
+	defer resp.Body.Close()
+	content, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, fmt.Errorf("could not read reasponse: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusNoContent && resp.StatusCode != http.StatusCreated {
+		return false, fmt.Errorf("%s; expected 201/204; got %d; body %s", item.Intake.ID.String(), resp.StatusCode, content)
+	}
+
+	var message string
+	if resp.StatusCode == http.StatusCreated {
+		message = "created"
+	} else {
+		message = "updated"
+	}
+	fmt.Fprintf(os.Stdout, "%s %s\n", message, item.Intake.LifecycleID.String)
+	return resp.StatusCode == http.StatusCreated, nil
```

Comment:
The backend will return 201 if a system is created and a 204 if one is updated. We use that value to report back to the user what the result of this upload was.